smk/QC_2.smk

```diff
@@ -1053,7 +1053,7 @@ SAMTOOLS_sort_perthread_memgb: 10
 # - I2
 #
 ILLUMINA:
-$(for i in {ilmn_samples}; do echo "- '$i'"; done)
+$(for i in {merged_illumina_samples} {ilmn_samples}; do echo "- '$i'"; done)
 
```

Comment:
Now, that the merged samples are added to ILLUMINA, is the user expected to remove the individual reps that were merged manually, or could it be done programmatically?

Reply:
For `mapping.yaml`, I think the right thing to do is to remove the individual reps. I have now implemented it (56c03e8).

For `assembly.yaml`, I have thought about it for a while but don't have a clear answer yet. It is tempting to remove it from the assembled sample list. But assembly is preceded by error-correction, where we went through a lot of pain to maintain all runs of a sample so that runs can be corrected individually. If we combine replicates into one fastq file, then we will be error-correcting for multiple-runs and multiple-samples. Are we making it worse? Does it matter? What do you think?

Reply:
I agree, if the samples to be merged are from biological replicates (different libraries), then the error correction should not be done on the merged fastq. IMO, if they are from bio replicates, they should not be "merged" in this way for tax. and func. profiling, either, but treated as separate samples. However, if they are from the same library, they should be treated as runs instead. 
For assembly, there would be multiple assemblies from the single, merged and the co-assembly with the combination of the singles and the merged. And the single assemblies can't even be reduced in numbers, because as you pointed out, the corrected files are needed from the non-merged fastqs, but the merged also need to be listed for downstream processes. Very messy!

Reply:
Agreed with the profiling comment. This is **only** for cases when the user decided to merge the replicates but not what we would recommend.

For coassembly, the merged-samples are not included and the replicates are kept. At least that part is not messy.

For other assemblies, if we keep **both** merged and single samples, we will have more assemblies than we need. Perhaps the user can decide what they want and edit the `assembly.yaml` file. But mapping during binning is non-redundant now that I have implemented the removal of replicates (56c03e8).

Reply:
>For coassembly, the merged-samples are not included and the replicates are kept.
Where does this happen, because the co-assembly inputs in the yaml are generated from the metadata file (L1066), and that should have merged and rep1+rep2+rep3 for the plotting steps. I think?

>But mapping during binning is non-redundant now that I have implemented the removal of replicates (https://github.com/arumugamlab/MIntO/commit/56c03e83c4ee8c4477e68e009f87e6fe823bf6eb).
binning_prep uses `assembly.yaml` for the depth file generation, not `mapping.yaml`. So assembly.yaml needs to be edited after running `assembly.smk` to remove the replicates.

I agree, that merging is not recommended **at all**.

Reply:
> Where does this happen, because the co-assembly inputs in the yaml are generated from the metadata file (L1066), and that should have merged and rep1+rep2+rep3 for the plotting steps. I think?

True. Only `'Full'` is non-redundant one. For `COAS_factor` and `SCL`-based coassemblies, both original reps and merged samples will be used.

> binning_prep uses assembly.yaml for the depth file generation, not mapping.yaml. So assembly.yaml needs to be edited after running assembly.smk to remove the replicates.

True, I missed that.

May be we can use the list that goes to `mapping.yaml` as a variable that defines which samples need tax-profiling and we can propagate that to sourmash clustering too? Then we can set the same list also for `assembly.yaml`, which makes it consistent across all steps. But with a clear note that **this will error-correct samples that may not be sequenced together**.

Reply:
>May be we can use the list that goes to mapping.yaml as a variable that defines which samples need tax-profiling and we can propagate that to sourmash clustering too?

I think it's good to do separate profiling on the replicates, as that would reveal how much they resemble each other.

I think we should leave it as it is now and leave the rest to the user. We could put some notes about manually handling the assembly part into the FAQ, with the `QC_2.yaml` text of the `MERGE_ILLUMINA_SAMPLES` section pointing to it.

Reply:
Cleaner implementation of `MERGE_ILLUMINA_SAMPLES` in da22e0d and 434c303.